src/coordination/coordinator_state_manager.cpp

```diff
@@ -31,37 +33,59 @@ CoordinatorStateManager::CoordinatorStateManager(CoordinatorInstanceInitConfig c
-  // Just return in-memory data in this example.
-  // May require reading from disk here, if it has been written to disk.
-  spdlog::trace("Loading cluster config");
+  MG_ASSERT(kv_store_, "Durability folder should be created.");
+  spdlog::trace("Loading cluster config from RocksDb");
+  auto servers = kv_store_->Get("servers");
+  if (!servers.has_value()) {
+    spdlog::trace("Didn't find anything stored on disk for cluster config.");
+    return cluster_config_;
+  }
+  spdlog::trace("Loading cluster config from disk.");
+  auto const json = nlohmann::json::parse(servers.value());
+  auto const real_servers = json.get<std::vector<std::tuple<int, std::string, std::string>>>();
+  cluster_config_->get_servers().clear();
+  for (auto const &real_server : real_servers) {
+    auto const &[coord_id, endpoint, aux] = real_server;
+    spdlog::trace("Recreating cluster config with id: {}, endpoint: {} and aux data: {} from disk.", coord_id, endpoint,
+                  aux);
+    auto one_server_config = cs_new<srv_config>(coord_id, 0, endpoint, aux, false);
+    cluster_config_->get_servers().push_back(std::move(one_server_config));
+  }
+
   return cluster_config_;
 }
 
 auto CoordinatorStateManager::save_config(cluster_config const &config) -> void {
-  // Just keep in memory in this example.
-  // Need to write to disk here, if want to make it durable.
+  MG_ASSERT(kv_store_, "Disk folder should be created");
   ptr<buffer> buf = config.serialize();
   cluster_config_ = cluster_config::deserialize(*buf);
-  spdlog::info("Saving cluster config.");
+  spdlog::info("Saving cluster config to disk.");
   auto servers = cluster_config_->get_servers();
+  std::vector<std::tuple<int, std::string, std::string>> servers_vec;
   for (auto const &server : servers) {
-    spdlog::trace("Server id: {}, endpoint: {}", server->get_id(), server->get_endpoint());
+    servers_vec.emplace_back(static_cast<int>(server->get_id()), server->get_endpoint(), server->get_aux());
+    spdlog::trace("Stored cluster config with id: {}, endpoint: {} and aux data: {} to disk.",
+                  static_cast<int>(server->get_id()), server->get_endpoint(), server->get_aux());
   }
+  nlohmann::json json(servers_vec);
+  kv_store_->Put("servers", json.dump());
 }
 
 auto CoordinatorStateManager::save_state(srv_state const &state) -> void {
-  // Just keep in memory in this example.
-  // Need to write to disk here, if want to make it durable.
+  // TODO(antoniofilipovic): Implement storing of server state to disk. For now
+  // as server state is just term and voted_for, we don't have to store it
+  spdlog::trace("Saving server state in coordinator state manager.");
   ptr<buffer> buf = state.serialize();
   saved_state_ = srv_state::deserialize(*buf);
 }
 
 auto CoordinatorStateManager::read_state() -> ptr<srv_state> {
-  // Just return in-memory data in this example.
-  // May require reading from disk here, if it has been written to disk.
```

Comment:
Is it true that `read_state` and `save_state` is for Raft logs and `save_config`, `load_config` for cluster membership?

Reply:
It is true what you said for `save_config` and `load_config`, they are for cluster membership. And as far as I can tell `read_state` and `save_state` is for `server_state` which holds info on term and what has been `voted_for`